Failure to find Perl breaks iOS xcode project generation

When I configure and generate an Xcode project to build for iOS I get the following warnings/errors from CMake:

```
Could NOT find Perl (missing: PERL_EXECUTABLE) 
CMake Warning at cmake/mkvk.cmake:14 (message):   Perl not found -> skipping mkvk target (this is harmless; only needed when   re-generating of vulkan headers and dfdutils is required) Call Stack (most recent call first):   CMakeLists.txt:22 (include)  
Could NOT find OpenGL (missing: OPENGL_gl_LIBRARY OPENGL_INCLUDE_DIR) 
Found MoltenVK.framwork at /Users/mark/vulkansdk-macos-1.2.141.1/MoltenVK/iOS/framework/MoltenVK.framework
Configuring done
CMake Error at CMakeLists.txt:167 (add_dependencies):   The dependency target "mkvk" of target "ktx_read" does not exist. Call Stack (most recent call first):   CMakeLists.txt:387 (commom_lib_settings)  
CMake Error at CMakeLists.txt:167 (add_dependencies):   The dependency target "mkvk" of target "ktx" does not exist. Call Stack (most recent call first):   CMakeLists.txt:386 (commom_lib_settings)  
Generating done
```

I also get a pop-up error message box telling me there was an error and the projects may not have been generated correctly. Indeed they were not. No .xcodeproj file was generated. I had to manually set `PERL_EXECUTABLE` in the CMake GUI, which was not easy because it only appears when _Advanced_ is checked, in order to generate the project. I have 2 versions of Perl installed, in /usr/bin and /opt/local/bin. When configuring for macOS, CMake automatically sets `PERL_EXECUTABLE` and it picks `/usr/bin/perl`.

2 things need fixing: (1) When it exists, Perl should be found when configuring for iOS and (2) if it is not found the project should still be successfully generated as the "harmless" comment in the first warning is true.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,10 @@ add_library( ktx_read ${LIB_TYPE}
 
 macro(commom_lib_settings lib write)
 
-    add_dependencies(${lib} mkvk)
+    if(TARGET mkvk)
+        # Creating vulkan headers is only required when Vulkan SDK updates.
+        add_dependencies(${lib} mkvk)
+    endif()
 
     set_target_properties(${lib} PROPERTIES
         PUBLIC_HEADER ${CMAKE_SOURCE_DIR}/include/ktx.h
@@ -385,9 +388,6 @@ set(KTX_BUILD_DIR "${CMAKE_BINARY_DIR}")
 
 commom_lib_settings(ktx 1)
 commom_lib_settings(ktx_read 0)
-
-## Creating vulkan headers is only required when Vulkan SDK updates.
-# add_dependencies(ktx mkvk)
 
 create_version_header(lib ktx)
 

